### PR TITLE
Restructure workflow tests

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phageannotator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phageannotator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: tests/main.nf.test [hunk 1]
```diff
@@ -29,22 +29,15 @@ nextflow_pipeline {
             assertAll(
                 { assert workflow.success },
                 { assert snapshot(
-                    path("${params.outdir}/VirusEnrichment"),
```
Comment (jfy133) [inline tests/main.nf.test:32]: Why are these removed? Have they been turned off in the defaul test?

File: workflows/phageannotator/main.nf [hunk 2]
```diff
@@ -295,42 +306,49 @@ workflow PHAGEANNOTATOR {
...
-    if ( !params.skip_bacphlip ) {
+    // if run_bacphlip == true, run subworkflow
+    if ( params.run_bacphlip ) {
         //
         // MODULE: Predict phage lifestyle using lysogeny specific genes
         //
         ch_bacphlip_lifestyle_tsv = BACPHLIP ( ch_anicluster_reps_fasta ).bacphlip_results
         ch_versions = ch_versions.mix( BACPHLIP.out.versions )
+
+        // TODO: Add ability to automatically add -meta flag to pharokka when multiple sequences are in input fasta
```
Comment (jfy133) [inline workflows/phageannotator/main.nf:340]: Addressed below? Does BACPHLIP also run prokka?

File: workflows/phageannotator/tests/run_instrain.nf.test [hunk 3]
```diff
@@ -0,0 +1,96 @@
+nextflow_workflow {
+
+    name "Test workflow: PHAGEANNOTATOR"
+    script "workflows/phageannotator/main.nf"
+    workflow "PHAGEANNOTATOR"
+
+    // Dependencies
+    tag "SEQKIT_SEQ"
+    //tag"APPENDSCREENHITS"
```
Comment (jfy133) [inline workflows/phageannotator/tests/run_instrain.nf.test]: Is it necessary to keep these comments them if the the test doesn't include them anymore? Or do you expect some changes?

File: workflows/phageannotator/tests/run_iphop.nf.test.snap [hunk 4]
```diff
@@ -64,94 +64,62 @@
...
                         },
-                        "test2.filtered.fna.gz:md5,741ab7b127375e64069553fa51b2eac9"
+                        "test2.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
                     ],
                     [
                         {
                             "id": "test"
                         },
-                        "test.filtered.fna.gz:md5,a3f4a97cd394993502c30dd444bc1b88"
+                        "test.filtered.fna.gz:md5,41b1f1f20079a5994f18090c6b562ad5"
```
Comment (jfy133) [inline workflows/phageannotator/tests/run_iphop.nf.test.snap:122]: Is it correct all the new MD5sums are identical? i.e. starting with 41b?

**Review by jfy133 — APPROVED** [review]
Only a summary review, but I don't see anything major :)

File: tests/main.nf.test (hunk 1 above)
Comment (CarsonJM) [reply tests/main.nf.test:32]: Yes they have been removed! I changed my default test to only run "essential" modules whereas it was previously a "full" test.

File: workflows/phageannotator/main.nf (hunk 2 above)
Comment (CarsonJM) [reply workflows/phageannotator/main.nf:340]: Not addressed yet. Currently I have the `-meta` argument included for all pipeline runs assuming that the user will almost always input multiple sequences. However, it is possible to run the pipeline on one sequence, so I want to add a function that will detect if the input FASTA has one or multiple sequences and adjust this parameter accordingly

File: workflows/phageannotator/tests/run_instrain.nf.test (hunk 3 above)
Comment (CarsonJM) [reply workflows/phageannotator/tests/run_instrain.nf.test]: Not necessary to keep, I will remove them!

File: workflows/phageannotator/tests/run_iphop.nf.test.snap (hunk 4 above)
Comment (CarsonJM) [reply workflows/phageannotator/tests/run_iphop.nf.test.snap:122]: Yes they should be identical. Samples `test1` and `test2` are actually the same contig FASTA because the other `b_fragilis` assemblies on test-datasets had too low of coverage when aligning the reads to pass some of the modules. But I want my pipeline tests to include multiple samples so I can make sure I don't have an accidental `queue` channel that should be a `value` channel
